stubs/core.stub

```diff
@@ -70,10 +70,16 @@ function bin2hex(string $string): string {}
 
 /**
  * @param array<mixed> $result
- * @param-out array<int|string, string|list<string>> $result
+ * @param-out array<int|string, string|list<string>|array<mixed>> $result
```

Comment:
`int` keys are only possible in php7 when using `parse_str($str);` without a second arg.
[per php manual this is highly discouraged](https://www.php.net/manual/en/function.parse-str.php) and was removed with php8:

> Warning
> Using this function without the result parameter is highly DISCOURAGED and DEPRECATED as of PHP 7.2. As of PHP 8.0.0, the result parameter is mandatory.

we could drop the `int` part from the array key to get a more precise type for the sugged use-cases, but would get errors for the discouraged use-case in return. wdyt?

Reply:
`string|list<string>|array<mixed>` - this doesn't make sense to me, that type is going to be normalized: https://phpstan.org/r/7006b26a-88ec-4766-ab84-bc4c8b93160a